### PR TITLE
test(sdk): Added data passing tests

### PR DESCRIPTION
**Pull request by Ark-kun**
Added data passing tests. This would be useful for checking the KFP-Tekton compatibility.

### CODE REVIEW

**Comment by Ark-kun** [comment]
The test passes: https://4e18c21c9d33d20f-dot-datalab-vm-staging.googleusercontent.com/#/runs/details/c47d946c-1401-459f-bfc4-07f1a2f70f18

**Comment by Bobgy** [comment]
/lgtm

Thanks @Ark-kun!

/Cc @chensun 
For review too

File: samples/test/data_passing.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    # Pass pipeline parameter; consume as value
+    consume_anything_as_value(anything_param)
+    consume_anything_as_value(something_param)
+    consume_anything_as_value(string_param)
+    consume_something_as_value(anything_param)
+    consume_something_as_value(something_param)
+    consume_string_as_value(anything_param)
+    consume_string_as_value(string_param)
+
+    # Pass pipeline parameter; consume as file
```
Comment (Bobgy) [inline samples/test/data_passing.py:120]: Shall we also adapt a pipeline revision that shows exactly which works in v2?

**Comment by chensun** [comment]
Thanks for putting up the test cases. 
This is related to https://github.com/kubeflow/pipelines/issues/5711#issuecomment-859769597
We'll use this PR to evaluate if we may address all or some of the data passing stories.

/assign @chensun

File: samples/test/data_passing.py (hunk 1 above)
Comment (Ark-kun) [reply samples/test/data_passing.py:120]: That's the next work. Check what's broken. Check which low hanging fruits can be fixed.

**Comment by Ark-kun** [comment]
Ideally, this pipeline should only have 6 lines - showing the 3*2=6 I/O combinations that KFP supports.

However, I felt that we need to add variants with different types, because, although KFP v1 is typed, but type-agnostic, there might be some other compilers may compile differently based on the particular types used. Ideally we might want to check not just `<Any>`, `Something` and `String`, but also `int`/`Integer`, `float`/`Float`, `bool`/`Boolean`, `list`/`JsonArray`, `dict`/`JsonObject` and their combinations with same type, `<Any>` an constant string arguments. However, this would have increased the pipeline size even more. Maybe it's still good idea to add later.

There are few more things not checked here: Using data of different types in `dsl.Condition` and `dsl.ParallelFor`.

**Comment by Ark-kun** [comment]
The tests are passing and this new test case is beneficial for accessing compatibility. 

/approve

**Pull request merged**